backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/TaskApi.java

```
@@ -14,47 +23,141 @@
+      MolgenisSession session = sessionManager.getSession(request);
+      String user = session.getSessionUser();
+      if (!"admin".equals(user)) {
+        throw new MolgenisException("Submit task failed: for now can only be done by 'admin");
+      }
+      String name = URLDecoder.decode(request.params("name"), StandardCharsets.UTF_8);
+      String parameters = request.body();
+      String id = taskService.submitTaskFromName(name, user, parameters);
+      return new TaskReference(id).toString();
+    }
+    throw new MolgenisException("Schema doesn't exist or permission denied");
+  }
+
+  private static Object getScript(Request request, Response response)
+      throws InterruptedException, UnsupportedEncodingException {
+    if (request.params("schema") == null || getSchema(request) != null) {
+      MolgenisSession session = sessionManager.getSession(request);
+      String user = session.getSessionUser();
+      if (!"admin".equals(user)) {
+        throw new MolgenisException("Submit task failed: for now can only be done by 'admin");
+      }
+      String name = URLDecoder.decode(request.params("name"), StandardCharsets.UTF_8.toString());
+      String parameters =
+          request.queryParams("parameters") != null
+              ? URLDecoder.decode(
+                  request.queryParams("parameters"), StandardCharsets.UTF_8.toString())
+              : null;
+      String id = taskService.submitTaskFromName(name, user, parameters);
+      // wait until done or timeout
+      int timeout = 0;
+      Task task = taskService.getTask(id);
+      // timeout a minute for these I would say?
+      while (task.isRunning() && timeout < 120) {
+        task = taskService.getTask(id);
+        timeout++;
+        Thread.sleep(500);
+      }
+      // get the output as bytes
+      return ((ScriptTask) task).getOutput();
+    }
+    throw new MolgenisException("Schema doesn't exist or permission denied");
```

move to to , fail fast instead of big if block

---

?